src/Backoffice/Flights/App/Requests/UpsertFlightRequest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Flights\App\Requests;
+
+use Carbon\CarbonImmutable;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+use Lightit\Backoffice\Flights\Domain\DataTransferObjects\FlightDTO;
+
+class UpsertFlightRequest extends FormRequest
+{
+    public function rules(): array
+    {
+        return [
+            'origin_id' => ['required', Rule::exists('cities', 'id')],
+            'destination_id' => ['required', Rule::exists('cities', 'id'), 'different:origin_id'],
+            'airline_id' => ['required', Rule::exists('airlines', 'id')],
```

Comment:
just asking, because this is the first PR i review on your challenge, do you already have created the relation between cities and airlines? if you do, you should also check if the cities are available for that airline

Reply:
I'm not sure about this, you need a relation between the cities and airlines to know if a certain airline can have a flight to a city. So that table is needed (because you can't save all the ids either in the city nor in the airline table)